2019-12-15 08:31:15,428 ERROR: Exception on /lzhang3/followers [GET] [in c:\users\lzhan\documents\github\nnhsse-flask\luke-home\microblog\venv\lib\site-packages\flask\app.py:1891]
Traceback (most recent call last):
  File "c:\users\lzhan\documents\github\nnhsse-flask\luke-home\microblog\venv\lib\site-packages\flask\app.py", line 2446, in wsgi_app
    response = self.full_dispatch_request()
  File "c:\users\lzhan\documents\github\nnhsse-flask\luke-home\microblog\venv\lib\site-packages\flask\app.py", line 1951, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "c:\users\lzhan\documents\github\nnhsse-flask\luke-home\microblog\venv\lib\site-packages\flask\app.py", line 1820, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "c:\users\lzhan\documents\github\nnhsse-flask\luke-home\microblog\venv\lib\site-packages\flask\_compat.py", line 39, in reraise
    raise value
  File "c:\users\lzhan\documents\github\nnhsse-flask\luke-home\microblog\venv\lib\site-packages\flask\app.py", line 1949, in full_dispatch_request
    rv = self.dispatch_request()
  File "c:\users\lzhan\documents\github\nnhsse-flask\luke-home\microblog\venv\lib\site-packages\flask\app.py", line 1935, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "c:\users\lzhan\documents\github\nnhsse-flask\luke-home\microblog\venv\lib\site-packages\flask_login\utils.py", line 261, in decorated_view
    return func(*args, **kwargs)
  File "C:\Users\lzhan\Documents\GitHub\nnhsse-flask\luke-home\microblog\app\routes.py", line 161, in followers
    followers=followers)
  File "c:\users\lzhan\documents\github\nnhsse-flask\luke-home\microblog\venv\lib\site-packages\flask\templating.py", line 140, in render_template
    ctx.app,
  File "c:\users\lzhan\documents\github\nnhsse-flask\luke-home\microblog\venv\lib\site-packages\flask\templating.py", line 120, in _render
    rv = template.render(context)
  File "c:\users\lzhan\documents\github\nnhsse-flask\luke-home\microblog\venv\lib\site-packages\jinja2\asyncsupport.py", line 76, in render
    return original_render(self, *args, **kwargs)
  File "c:\users\lzhan\documents\github\nnhsse-flask\luke-home\microblog\venv\lib\site-packages\jinja2\environment.py", line 1008, in render
    return self.environment.handle_exception(exc_info, True)
  File "c:\users\lzhan\documents\github\nnhsse-flask\luke-home\microblog\venv\lib\site-packages\jinja2\environment.py", line 780, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "c:\users\lzhan\documents\github\nnhsse-flask\luke-home\microblog\venv\lib\site-packages\jinja2\_compat.py", line 37, in reraise
    raise value.with_traceback(tb)
  File "C:\Users\lzhan\Documents\GitHub\nnhsse-flask\luke-home\microblog\app\templates\_follower.html", line 14, in template
    {% if user.is_following(follower)}
  File "c:\users\lzhan\documents\github\nnhsse-flask\luke-home\microblog\venv\lib\site-packages\jinja2\environment.py", line 1005, in render
    return concat(self.root_render_func(self.new_context(vars)))
  File "C:\Users\lzhan\Documents\GitHub\nnhsse-flask\luke-home\microblog\app\templates\followers.html", line 14, in root
  File "C:\Users\lzhan\Documents\GitHub\nnhsse-flask\luke-home\microblog\app\templates\base.html", line 14, in root
    <span class="icon-bar"></span>
  File "c:\users\lzhan\documents\github\nnhsse-flask\luke-home\microblog\venv\lib\site-packages\flask_bootstrap\templates\bootstrap\base.html", line 10, in root
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
  File "c:\users\lzhan\documents\github\nnhsse-flask\luke-home\microblog\venv\lib\site-packages\flask_bootstrap\templates\bootstrap\base.html", line 20, in block_doc
    {% block body -%}
  File "c:\users\lzhan\documents\github\nnhsse-flask\luke-home\microblog\venv\lib\site-packages\flask_bootstrap\templates\bootstrap\base.html", line 39, in block_html
  File "c:\users\lzhan\documents\github\nnhsse-flask\luke-home\microblog\venv\lib\site-packages\flask_bootstrap\templates\bootstrap\base.html", line 92, in block_body
  File "C:\Users\lzhan\Documents\GitHub\nnhsse-flask\luke-home\microblog\app\templates\base.html", line 81, in block_content
  File "C:\Users\lzhan\Documents\GitHub\nnhsse-flask\luke-home\microblog\app\templates\followers.html", line 26, in block_app_content
  File "c:\users\lzhan\documents\github\nnhsse-flask\luke-home\microblog\venv\lib\site-packages\jinja2\environment.py", line 830, in get_template
    return self._load_template(name, self.make_globals(globals))
  File "c:\users\lzhan\documents\github\nnhsse-flask\luke-home\microblog\venv\lib\site-packages\jinja2\environment.py", line 804, in _load_template
    template = self.loader.load(self, name, globals)
  File "c:\users\lzhan\documents\github\nnhsse-flask\luke-home\microblog\venv\lib\site-packages\jinja2\loaders.py", line 125, in load
    code = environment.compile(source, name, filename)
  File "c:\users\lzhan\documents\github\nnhsse-flask\luke-home\microblog\venv\lib\site-packages\jinja2\environment.py", line 591, in compile
    self.handle_exception(exc_info, source_hint=source_hint)
  File "c:\users\lzhan\documents\github\nnhsse-flask\luke-home\microblog\venv\lib\site-packages\jinja2\environment.py", line 780, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "c:\users\lzhan\documents\github\nnhsse-flask\luke-home\microblog\venv\lib\site-packages\jinja2\_compat.py", line 37, in reraise
    raise value.with_traceback(tb)
  File "C:\Users\lzhan\Documents\GitHub\nnhsse-flask\luke-home\microblog\app\templates\_follower.html", line 14, in template
    {% if user.is_following(follower)}
  File "c:\users\lzhan\documents\github\nnhsse-flask\luke-home\microblog\venv\lib\site-packages\jinja2\environment.py", line 497, in _parse
    return Parser(self, source, name, encode_filename(filename)).parse()
  File "c:\users\lzhan\documents\github\nnhsse-flask\luke-home\microblog\venv\lib\site-packages\jinja2\parser.py", line 901, in parse
    result = nodes.Template(self.subparse(), lineno=1)
  File "c:\users\lzhan\documents\github\nnhsse-flask\luke-home\microblog\venv\lib\site-packages\jinja2\parser.py", line 883, in subparse
    rv = self.parse_statement()
  File "c:\users\lzhan\documents\github\nnhsse-flask\luke-home\microblog\venv\lib\site-packages\jinja2\parser.py", line 130, in parse_statement
    return getattr(self, 'parse_' + self.stream.current.value)()
  File "c:\users\lzhan\documents\github\nnhsse-flask\luke-home\microblog\venv\lib\site-packages\jinja2\parser.py", line 211, in parse_if
    node.test = self.parse_tuple(with_condexpr=False)
  File "c:\users\lzhan\documents\github\nnhsse-flask\luke-home\microblog\venv\lib\site-packages\jinja2\parser.py", line 620, in parse_tuple
    args.append(parse())
  File "c:\users\lzhan\documents\github\nnhsse-flask\luke-home\microblog\venv\lib\site-packages\jinja2\parser.py", line 612, in <lambda>
    parse = lambda: self.parse_expression(with_condexpr=False)
  File "c:\users\lzhan\documents\github\nnhsse-flask\luke-home\microblog\venv\lib\site-packages\jinja2\parser.py", line 433, in parse_expression
    return self.parse_or()
  File "c:\users\lzhan\documents\github\nnhsse-flask\luke-home\microblog\venv\lib\site-packages\jinja2\parser.py", line 450, in parse_or
    left = self.parse_and()
  File "c:\users\lzhan\documents\github\nnhsse-flask\luke-home\microblog\venv\lib\site-packages\jinja2\parser.py", line 459, in parse_and
    left = self.parse_not()
  File "c:\users\lzhan\documents\github\nnhsse-flask\luke-home\microblog\venv\lib\site-packages\jinja2\parser.py", line 470, in parse_not
    return self.parse_compare()
  File "c:\users\lzhan\documents\github\nnhsse-flask\luke-home\microblog\venv\lib\site-packages\jinja2\parser.py", line 474, in parse_compare
    expr = self.parse_math1()
  File "c:\users\lzhan\documents\github\nnhsse-flask\luke-home\microblog\venv\lib\site-packages\jinja2\parser.py", line 496, in parse_math1
    left = self.parse_concat()
  File "c:\users\lzhan\documents\github\nnhsse-flask\luke-home\microblog\venv\lib\site-packages\jinja2\parser.py", line 507, in parse_concat
    args = [self.parse_math2()]
  File "c:\users\lzhan\documents\github\nnhsse-flask\luke-home\microblog\venv\lib\site-packages\jinja2\parser.py", line 517, in parse_math2
    left = self.parse_pow()
  File "c:\users\lzhan\documents\github\nnhsse-flask\luke-home\microblog\venv\lib\site-packages\jinja2\parser.py", line 528, in parse_pow
    left = self.parse_unary()
  File "c:\users\lzhan\documents\github\nnhsse-flask\luke-home\microblog\venv\lib\site-packages\jinja2\parser.py", line 547, in parse_unary
    node = self.parse_postfix(node)
  File "c:\users\lzhan\documents\github\nnhsse-flask\luke-home\microblog\venv\lib\site-packages\jinja2\parser.py", line 676, in parse_postfix
    node = self.parse_call(node)
  File "c:\users\lzhan\documents\github\nnhsse-flask\luke-home\microblog\venv\lib\site-packages\jinja2\parser.py", line 793, in parse_call
    self.stream.expect('rparen')
  File "c:\users\lzhan\documents\github\nnhsse-flask\luke-home\microblog\venv\lib\site-packages\jinja2\lexer.py", line 388, in expect
    next(self)
  File "c:\users\lzhan\documents\github\nnhsse-flask\luke-home\microblog\venv\lib\site-packages\jinja2\lexer.py", line 359, in __next__
    self.current = next(self._iter)
  File "c:\users\lzhan\documents\github\nnhsse-flask\luke-home\microblog\venv\lib\site-packages\jinja2\lexer.py", line 562, in wrap
    for lineno, token, value in stream:
  File "c:\users\lzhan\documents\github\nnhsse-flask\luke-home\microblog\venv\lib\site-packages\jinja2\lexer.py", line 683, in tokeniter
    filename)
jinja2.exceptions.TemplateSyntaxError: unexpected '}'
2019-12-15 08:31:44,100 INFO: Microblog startup [in C:\Users\lzhan\Documents\GitHub\nnhsse-flask\luke-home\microblog\app\__init__.py:45]
2019-12-15 08:33:19,859 INFO: Microblog startup [in C:\Users\lzhan\Documents\GitHub\nnhsse-flask\luke-home\microblog\app\__init__.py:45]
